### DISCUSSION TITLE
Filter Icon disapear

### DISCUSSION THREAD

**Discussion by methos02** [orchidsoftware/platform#2685] [Q&A]
Hello, 
I update Orchid from 13 to 14. I also update the filter code and everything works except that the button's icon disapear.

`
<button class="btn btn-sm btn-link dropdown-toggle p-0 me-1" type="button" data-bs-toggle="dropdown" aria-haspopup="true" data-bs-boundary="viewport" aria-expanded="false">
            </button>
`

It's the same with the filter badge, there are no cross to close it. Is it because i use fontawesome icon in my config ?

Thx in advance

LEON Frédéric

**Comment by methos02** [comment]
I found a pretty dirty solution for the moment :s 
`
(function() {
    const pushState = history.pushState;
    const replaceState = history.replaceState;

    history.pushState = function() {
        pushState.apply(history, arguments);
        window.dispatchEvent(new Event('pushState'));
        window.dispatchEvent(new Event('locationChange'));
    };

    history.replaceState = function() {
        replaceState.apply(history, arguments);
        window.dispatchEvent(new Event('replaceState'));
        window.dispatchEvent(new Event('locationChange'));
    };

    window.addEventListener('popstate', function() {
        window.dispatchEvent(new Event('locationChange'))
    });
})();


let interval_id;
loadImages();
window.addEventListener('locationChange', function() {
    loadImages();
});

function loadImages() {
    interval_id = setInterval(async () => {
        if (document.querySelector('[data-controller="table"]') !== null) {
            const badges = document.querySelectorAll('a[data-filter]');
            if(badges !== null) badges.forEach(badge => badge.insertAdjacentHTML('beforeend', '<button style="border: none; background-color: transparent; padding: 0 0 3px 5px;">x</button>'))

            const filter_buttons = document.querySelectorAll('[data-controller="filter"] > button');
            if(filter_buttons !== null) filter_buttons.forEach(button => {
                button.style.textDecoration = "none";
                button.insertAdjacentHTML('beforeend', '<i class="fa-solid fa-filter"></i>');
            });

            clearInterval(interval_id);
        }
    }, 10)
}
`

**Comment by tabuna** [comment]
It's hard for me to tell, since there are no steps to reproduce. 
Did you clear the cache for the templates?

```bash
artisan view:clear
```

**Reply by methos02** [reply]
Thx for the answer, but yes I did it.  I ask my self why this button is empty and how the icon is generate.

**Reply by methos02** [reply]
By the way, I have this problem on two differents projects :s

**Comment by lucatrev** [comment]
same scenario and same problem here:

- upgrade 13 -> 14
- followed all guide steps (both filter code and icons)
- `artisan view:clear` as suggested

still no filter icon near table headers, so no able to filter table rows. As minor bug also `x` for filter badge is gone but i can remove the badge by clicking

**More context**:

Switching back to bs icons fixes the issue but misses other icons:

```diff
    // config/platform.php
    'icons' => [
-        'orc' => \Orchid\IconPack\Path::getFolder(),
+        'bs'  => \Orchid\Support\BootstrapIconsPath::getFolder(),
```
and this fixes everything:
```diff
    // config/platform.php
    'icons' => [
        'orc' => \Orchid\IconPack\Path::getFolder(),
+        'bs'  => \Orchid\Support\BootstrapIconsPath::getFolder(), // of course here you can add fontawesome reference for icons
```

**Comment by darrylk** [comment]
I have what may be a related problem but not exactly the same.  I thought it might overlap so I'm commenting here.

I'm running Orchid 14.8.1 (using "orchid/icons": "^2.0") and made code changes to existing classes to add a filter icon beside the 'uniqueid' column heading "Reference Id".  I followed Orchid doc page "Sorting and Filtering In a Table" at https://orchid.software/en/docs/quickstart-sort-filter-table/#docs.

The column heading "Reference Id" is displayed but there is no filter icon beside it as I had expected.

Also, there are filter statements present in the page source within the <th> tag for this column, but for some reason no icon is displayed.  Perhaps it's evident from the html why it isn't being displayed but I couldn't figure it out.  Here's a screenshot of the html for the 'uniqueid' column.  I'd be glad to share the appropriate class sections if this doesn't provide enough information.

![image](https://github.com/orchidsoftware/platform/assets/5773690/4b741192-8e03-49e4-b5ff-c8d16d211220)

I also found that there's a small white space to the left of the Reference Id heading where you can click and get the filter input display.
So just the icon is missing. Might have to do with using orchid icons?

![image](https://github.com/orchidsoftware/platform/assets/5773690/cbad7b56-3913-479f-bf2e-1b8be8006050)

**Reply by darrylk** [reply]
Turns out if I remove the orchid icon config statment like so:
/*
    'icons' => [
        'orc' => \Orchid\IconPack\Path::getFolder(),
    ],
*/
the filter icon appears.
![image](https://github.com/orchidsoftware/platform/assets/5773690/9ffa0d7f-716a-4786-b9e2-672bf1248f65)
But some of the icons in use elsewhere disappear.
For example, the Back to Maps icon below (->icon("directions")) disappears.
![image](https://github.com/orchidsoftware/platform/assets/5773690/87968a9d-0003-4144-a563-100f7186bc5d)
![image](https://github.com/orchidsoftware/platform/assets/5773690/11750418-12d6-4f1c-946d-f318ea832388)
